src/braket/default_simulator/openqasm/_helpers/casting.py

```diff
@@ -116,19 +117,21 @@ def _(into: UintType, variable: LiteralType) -> IntegerLiteral:
 @cast_to.register
 def _(into: FloatType, variable: LiteralType) -> FloatLiteral:
     """Cast to float"""
+    value = variable.value.evalf() if isinstance(variable, SymbolLiteral) else variable.value
     if into.size is None:
-        return FloatLiteral(float(variable.value))
-    if into.size.value not in (16, 32, 64):
-        raise ValueError("Float size must be one of {16, 32, 64}.")
-    value = float(np.array(variable.value, dtype=np.dtype(f"float{into.size.value}")))
+        value = float(value)
+    else:
+        if into.size.value not in (16, 32, 64):
+            raise ValueError("Float size must be one of {16, 32, 64}.")
+        value = float(np.array(value, dtype=np.dtype(f"float{into.size.value}")))
```

Comment:
Why the change back here?

Reply:
hadn't pulled in the changes when I started this code changes and `variable.value` -> `value` created a conflict and I ended up just selecting one half of the diff to resolve. I can change back, but I also don't mind factoring out the common logic of wrapping in a FloatLiteral as written